src/scripts/components/layer/wv.layer.list.js

```diff
@@ -0,0 +1,120 @@
+import React from 'react';
+import LayerRadio from './wv.layer.radio.js';
+import { CellMeasurer, CellMeasurerCache, List } from 'react-virtualized';
+
+/*
+ * A react component, Builds a list of layers using the LayerRadio component
+ * @class LayerList
+ * @extends React.Component
+ */
+export default class LayerList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      layerFilter: props.config.layerOrder,
+      active: props.model.active,
+      infoExpanded: [],
+      width: props.width,
+      height: props.height
+    };
+    this._cache = new CellMeasurerCache({
+      fixedWidth: true,
+      minHeight: 23,
+      minWidth: this.state.width
+    });
+
+    this._rowRenderer = this._rowRenderer.bind(this);
+    this._setListRef = this._setListRef.bind(this);
+    this.saveExpandedInfoState = this.saveExpandedInfoState.bind(this);
+  }
+  componentWillUpdate(){
+    this._cache.clearAll();
+  }
+  /*
+   * Saves the visibility/state of the metadata for each layer
+   * @method saveExpandedInfoState
+   * @param {string} layerID - the layer which has metadata that
+   *  needs to be toggled
+   * @return {void}
+   */
+  saveExpandedInfoState(layer){
+    var { infoExpanded } = this.state;
+    var index = infoExpanded.indexOf(layer);
+    if(index > -1){
+      infoExpanded.splice(index, 1); // Removes layer from expanded list
+    } else {
+      infoExpanded.push(layer);
+    }
+    this.setState({ infoExpanded: infoExpanded });
+  }
+  /*
+   * Recalculates the row height for a given rowIndex
+   * @method recalculateRowHeight
+   * @param {number} index - Index of the row to be recalculated
+   * @return {void}
+   */
+  recalculateRowHeight (rowIndex){
+    this._cache.clear(rowIndex, 0);
+    this._layerList.recomputeRowHeights(rowIndex);
+    this._layerList.scrollToRow(rowIndex);
+  }
+  _rowRenderer ({ index, isScrolling, key, parent, style }) {
+    var enabled = false;
+    for(var i of this.props.model.active){
+      if(i.id === this.state.layerFilter[index])
```

Comment:
Could we use `this.props.config.layerOrder[index]` instead?

Or this if it's more readable;
```
var { layerOrder: layerFilter } = this.props.config;
layerFilter[index];
```